Add 'model_predict' programmatic interface (API)

We will create the necessary api to allow users to generate an svm prediction (using a model stored in a NoSQL cache), via a programmatic-interface, instead of a standalone web HTML5 interface.

We run our defined unit-test in pytest_session.py, from a dedicated terminal:

```bash
vagrant@vagrant-ubuntu-trusty-64:/vagrant/test$ py.test
============================= test session starts ==============================

platform linux2 -- Python 2.7.6, pytest-2.8.0, py-1.4.30, pluggy-0.3.1
rootdir: /vagrant/test, inifile: pytest.ini
collected 3 items

programmatic_interface/pytest_session.py ..F

=================================== FAILURES ===================================

_____________________________ check_model_generate _____________________________


    def check_model_generate():
        """@check_model_generate

        This method tests the 'model_generate' session.

        """

>       assert requests.post(endpoint_url, headers=headers, data=get_sample_json
('sample-model-generate.json'))
E       assert <Response [500]>
E        +  where <Response [500]> = <function post at 0x7ffb44ea0f50>('http://localhost:5000/load-data/', headers={'Content-Type': 'application/json'}, data='{"properties": {"svm_session_id": "1", "svm_model_type": "classification", "svm_session": "model_generate"}}')
E        +    where <function post at 0x7ffb44ea0f50> = requests.post
E        +    and   '{"properties": {"svm_session_id": "1", "svm_model_type": "classification", "svm_session": "model_generate"}}' = get_sample_json('sample-model-generate.json')

programmatic_interface/pytest_session.py:73: AssertionError
====================== 1 failed, 2 passed in 0.55 seconds ======================
```

then, we inspect the corresponding output, on the second terminal window, manually running `app.py`:

```bash
127.0.0.1 - - [07/Oct/2015 14:07:09] "POST /load-data/ HTTP/1.1" 500 -
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1836, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1820, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1403, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/vagrant/interface/views.py", line 27, in load_data
    dataset  = request.get_json()['dataset']
KeyError: 'dataset'
```

--- a/brain/load_data.py
+++ b/brain/load_data.py
@@ -141,7 +141,7 @@ class Load_Data(object):
             session.generate_model()
 
         # return
-        if session.return_error:
+        if session.return_error():
             return False
         else:
             return 'Model properly generated'

--- a/interface/static/data/json/programmatic_interface/sample-model-generate.json
+++ b/interface/static/data/json/programmatic_interface/sample-model-generate.json
@@ -2,6 +2,6 @@
     "properties": {
         "svm_session": "model_generate",
         "svm_session_id": "1",
-        "svm_model_type": "classification",
+        "svm_model_type": "classification"
     }
 }

--- a/interface/views.py
+++ b/interface/views.py
@@ -24,7 +24,10 @@ def load_data():
         # load programmatic-interface
         if request.get_json():
             # get necessary components from the dataset
-            dataset  = request.get_json()['dataset']
+            if 'dataset' in request.get_json():
+                dataset = request.get_json()['dataset']
+            else:
+                dataset = None
             settings = request.get_json()['properties']
 
             # restructure the dataset

--- a/test/programmatic_interface/pytest_session.py
+++ b/test/programmatic_interface/pytest_session.py
@@ -62,3 +62,12 @@ def check_data_append():
     """
 
     assert requests.post(endpoint_url, headers=headers, data=get_sample_json('sample-data-append.json'))
+
+def check_model_generate():
+    """@check_model_generate
+
+    This method tests the 'model_generate' session.
+
+    """
+
+    assert requests.post(endpoint_url, headers=headers, data=get_sample_json('sample-model-generate.json'))

